internal/httputil/httputil.go

```diff
@@ -6,48 +6,40 @@ import (
 	"net/http"
 	"os"
 	"path/filepath"
-	"strings"
 	"time"
 )
 
-func LoadCerts(caDir string) (string, error) {
+func NewCertPool(caDir string) (*x509.CertPool, error) {
+	caCertPool, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, err
+	}
 	if caDir == "" {
-		return "", nil
+		return caCertPool, nil
 	}
 
-	certs := []string{}
 	dirEntries, err := os.ReadDir(caDir)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	for _, e := range dirEntries {
 		if e.IsDir() {
 			continue
 		}
 		data, err := os.ReadFile(filepath.Join(caDir, e.Name()))
 		if err != nil {
-			return "", err
+			return nil, err
 		}
-		certs = append(certs, string(data))
+		// ignore ok - which just indicates if any cert was added
+		// This will ignore non-certs, which will return !ok
+		caCertPool.AppendCertsFromPEM(data)
```

Comment:
If we get `!ok`, should we error instead of silently ignoring?

Seems like we could construct a fairly nice error message like "failed to add certs from <filename>: invalid certificate data".

A cert that missed a trailing character or that accidentally included an errant whitespace character would be considered a non-cert by this code and would make troubleshooting a lot more difficult.

Reply:
The problem is that we can't tell a "badly formatted cert" from a "someone put a text file in here that we should ignore". I would say we should log an error, if anything, not error out.